server/modules/module_library.js

```diff
@@ -15,15 +15,25 @@ limitations under the License.
 
 'use strict';
 
-var EventEmitter = require('events');
-var assert = require('assert');
+const EventEmitter = require('events');
+const ModuleDef = require('server/modules/module_def');
+const assert = require('assert');
+
+class EmptyModuleDef extends ModuleDef {
+  constructor() {
+    super('_empty');
+    // TODO(applmak): ^ this hacky.
+    // However, b/c of the hack, this module will never become valid.
+    this.whenLoadedPromise = Promise.resolve(this);
```

Comment:
Can you explain this?

Reply:
Well, we need to transition to something, so this is that something. It's better than abusing solid.js for this purpose, IMO.

Reply:
No, I mean technically, what does whenLoadedPromise do...  How are you using it?  How does that work?  I've never seen code like that before.  Is the object thenable?  What's the point?

Reply:
It's a promise that is resolved when the module source is loaded, parsed, and verified. It's normally set in checkValidity_(). It's reset to a new unresolved promise whenever we're in the middle of loading the file, but before we've verified it. If the file fails to parse, or any other error occurs, we reject the promise.

In this case, we want to pretend that this module is totally legit, so we create a promise that's already resolved to this instance of the module, so that things that chain onto the end get the right value.